test/lifecycle_test.dart

```diff
@@ -134,6 +134,15 @@ main() {
         isComponent2: true,
       );
 
+      test('initializes correctly using initializeState', () {
```

Comment:
The existing tests using `this.state = ` should be updated to use this new API :
> `test('initializes state with the value set in `init`', () {`
> `test('throws when state is initialized in both init and getInitialState', () {`

After that's done, I don't think we'll need this new test anymore.

The `expect(component.state, isA<JsBackedMap>());` is a great addition, though! I think that should be added to the existing tests.